src/pkcs11/mechanism.c

```diff
@@ -1536,6 +1541,10 @@ sc_pkcs11_decrypt(sc_pkcs11_operation_t *operation,
 	if (pulDataLen)
 		*pulDataLen = ulDataLen;
 
+	/* Skip DecryptFinalize for PKCS#1 v1.5 padding to prevent time side-channel leakage */
+	if (((CK_MECHANISM_PTR) &operation->mechanism)->mechanism == CKM_RSA_PKCS)
```

Comment:
nit: Cast type should be adjacent to the variable:
```suggestion
	if (((CK_MECHANISM_PTR)&operation->mechanism)->mechanism == CKM_RSA_PKCS)
```